protecode/util.py

```diff
@@ -141,59 +235,22 @@ def iter_artefact_metadata(
                 data=vulnerability_finding,
                 discovery_date=discovery_date,
             )
-            has_vulnerability_findings = True
-
-    if not has_license_findings:
-        # yield dummy finding so that delivery-service can handle "no license findings"-finding
-        meta = dso.model.Metadata(
-            datasource=dso.model.Datasource.BDBA,
-            type=dso.model.Datatype.LICENSE,
-            creation_date=now,
-            last_update=now,
-        )
 
-        dummy_finding = dso.model.LicenseFinding(
-            id=package_id,
-            scan_id=scan_id,
-            severity=gcm.Severity.NONE.name,
-            license=license,
-        )
-
-        yield dso.model.ArtefactMetadata(
-            artefact=artefact_ref,
-            meta=meta,
-            data=dummy_finding,
-            discovery_date=discovery_date,
-        )
-
-    if not has_vulnerability_findings:
-        # yield dummy finding so that delivery-service can handle "no license findings"-finding
-        meta = dso.model.Metadata(
-            datasource=dso.model.Datasource.BDBA,
-            type=dso.model.Datatype.VULNERABILITY,
-            creation_date=now,
-            last_update=now,
-        )
-
-        dummy_finding = dso.model.VulnerabilityFinding(
-            id=dso.model.BDBAPackageId(
-                package_name=None,
-                package_version=None,
-            ),
-            scan_id=scan_id,
-            severity=gcm.Severity.NONE.name,
-            cve=None,
-            cvss_v3_score=-1,
-            cvss=None,
-            summary=None,
-        )
-
-        yield dso.model.ArtefactMetadata(
-            artefact=artefact_ref,
-            meta=meta,
-            data=dummy_finding,
-            discovery_date=discovery_date,
-        )
+            # remove vuln finding because it is still relevant and must not be deleted afterwards
```

Comment:
might be made slightly more self-describing by moving code from condition into a function and call it in list-comprehension